app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -130,9 +133,8 @@ public void run() {
                             }
                         });
                     }
-                } catch (HttpResponseException e) {
-                    if (e.getStatusCode() == 400) {
-                        String msg = e.getMessage();
+                    if (httpResponse.code() == 400) {
```

Comment:
may need to check the response code for success earlier, since it no longer throws an error.